### ISSUE TITLE
WZA wymagania metod

### ISSUE THREAD

**Issue by matcapi** [allegro/allegro-api#8136]
Dzień dobry,

Chciałbym się dowiedzieć w jaki sposób mogę uzyskać następujące informacje odnośnie metod wysyłki w WZA:
- wymagany typ (paczka/koperta)
- jakiej długości może być komentarz do przesyłki
- czy podanie kwoty ubezpieczenia jest wymagane

Chciałbym również spytać czy którakolwiek metod w WZA zamiast wymierów (wysokość, szerokość, długość), wymaga w request body podania gabarytu (A, B, C)?

Pozdrawiam

**Comment by Lukasz-Zurek** [comment]
Jeżeli chodzi o:

- wymagany typ przesyłki - zwracamy go dla usług dostawy, gdy skorzystasz z [GET /shipment-management/delivery-services](https://developer.allegro.pl/documentation/#operation/getDeliveryServices), w polu **"packageTypes"**. Dostępne wartości: "DOX", "PACKAGE", "PALLET", "OTHER",
- długość pola "description" - zależna jest od przewoźnika. Nie określamy tego w dokumentacji, a i limity te mogą być zmienne, np.
    - DPD, Orlen, InPost, Fedex, GLS: 100 znaków, 
    - UPS: 50, 
    - Poczta Polska: 40, 
    - DHL: 30,
 - przekazanie kwoty i waluty dla ubezpieczenia w przypadku korzystania z zasobu [POST /shipment-management/shipments/create-commands](https://developer.allegro.pl/documentation/#operation/createNewShipment) - jest wymagane.

Nie wymagamy podawania gabarytu - bazujemy tutaj tylko na wymiarach paczki.

**Comment by matcapi** [comment]
Dziękuję za odpowiedź. 
Gdy na sandbox korzystam z [GET /shipment-management/delivery-services](https://developer.allegro.pl/documentation/#operation/getDeliveryServices), to dostaję:
 "code": "AccessDenied"
 "message": "Access is denied"
 "details": null
 "path": null
 "userMessage": "Brak dostępu do danego zasobu."

 trace-id:445f318aaf2e1cdd
 
 Czy problem leży po stronie konfiguracji konta?

**Comment by Lukasz-Zurek** [comment]
Przyczyna jest inna. Twój token nie posiada scope'a (uprawnienia): **allegro:api:shipments:write**, który wymagany jest dla tego zasobu.

Możesz je dodać w [ustawieniach aplikacji](https://apps.developer.allegro.pl.allegrosandbox.pl). Konieczna będzie jednak ponowna autoryzacja.

**Comment by matcapi** [comment]
W dokumentacji [GET /shipment-management/delivery-services](https://developer.allegro.pl/documentation/#operation/getDeliveryServices) jest napisane: 
[bearer-token-for-user](https://developer.allegro.pl/documentation/#section/Authentication/bearer-token-for-user) (allegro:api:shipments:write)

Jednak gdy próbuję zalogować się z tym scopem jako user, to dostaję:
"error": "invalid_scope"
"error_description": "Invalid scope: allegro:api:shipments:read"

Czy ten endpoint jest dostępny tylko dla [bearer-token-for-application](https://developer.allegro.pl/documentation/#section/Authentication/bearer-token-for-application)?

**Comment by Lukasz-Zurek** [comment]
Nie, ten endpoint zgodnie z dokumentacją, dostępny jest dla _bearer-token-for-user_.

Sprawdzałem logi i ostatni z godziny 11:50:32 nadal wskazuje na to, że nie masz **allegro:api:shipments:write**. Ponadto nie dodałeś wspomnianego scope'a w [ustawieniach aplikacji](https://apps.developer.allegro.pl.allegrosandbox.pl/).

**Comment by matcapi** [comment]
Masz rację, poprawiłem konfigurację i działa. Dziękuję za pomoc.

**Comment by kdabek** [comment]
Dołączę do pytania, bo mam podobny problem i dostaję w response odpowiedź "Opis przesyłki jest za długi".
Czy można w takim razie dołączyć do pola `details` w response jaki rozmiar jest maksymalny dla danego przewoźnika? 
Bo teraz to trochę zgaduj zgadula

**Comment by Lukasz-Zurek** [comment]
@kdabek Dziękuję. Twoja sugestia jest jak najbardziej trafna, z tym, że realizacja tego byłaby utrudniona, ponieważ limity te mogą się zmieniać niezależnie od nas i może się zdarzyć, że komunikat w pewnym momencie nie odzwierciedla stanu faktycznego. Na ten moment obowiązują jednak limity takie jak wskazałem powyżej.
